src/App.tsx

```diff
@@ -1,24 +1,27 @@
-import React from "react";
+import React, { useEffect } from "react";
 import "./assets/styles/main.scss"
+import { Footer } from "./components/Footer";
+import { fetchMovies } from "./store/reducers/movieReducer";
+import { useDispatch, useSelector } from "react-redux";
 import { MainBanner } from "./components/Banner";
+import { DetailedMovieInfoWrapper } from "./containers/DetailedMovieInfoWrapper";
 import { MoviesContainer } from "./containers/MoviesContainer";
-import { Footer } from "./components/Footer";
-import {DialogProvider} from "./context/DialogContext";
-import {Modal} from "./components/Modal";
-
+import { Modals } from "./containers/Modals";
 
 export const App: React.FunctionComponent = () => {
-    return (
-        <React.StrictMode>
-            <DialogProvider>
-                <MainBanner />
-                <MoviesContainer />
-                <Footer />
+    const dispatch = useDispatch()
+    const viewMode = useSelector((state: any) => state.appData.viewMode)
+
+    useEffect(() => {
+        dispatch(fetchMovies())
+    }, [])
 
-                <Modal type="add"/>
-                <Modal type="edit"/>
-                <Modal type="delete"/>
-            </DialogProvider>
-        </React.StrictMode>
+    return (
+        <>
+            { viewMode === "search" ? <MainBanner /> : <DetailedMovieInfoWrapper /> }
```

Comment:
I would say, having some static component (e.g. `<Header />`) which renders its content based on state, would be more appropriate from reconciliation standpoint.